web/src/ui_util.ts

```diff
@@ -40,8 +40,7 @@ export function update_unread_count_in_dom($unread_count_elem: JQuery, count: nu
     // This function is used to update unread count in top left corner
     // elements.
     const $unread_count_span = $unread_count_elem.find(".unread_count");
-
-    if (count === 0) {
+    if (!count) {
```

Comment:
What's the story with this change? I would expect based on the above we're still passing an integer to this.

Reply:
It was giving undefined in some cases but after you rewrote the logic it is giving an integer every time. hence, reverted this change in the updated PR.